prismjs/prismjs-tests.ts

```diff
@@ -8,3 +6,19 @@ Prism.highlightElement(element, false);
 Prism.highlightAll(true, callback);
 Prism.highlightAll(true);
 
+const hookCallback: PrismJS.HookCallback = env => null;
+Prism.hooks.add("before-highlightall", hookCallback);
+Prism.hooks.add("future-hook", hookCallback);
+
+var language = "js";
+var tokens = Prism.tokenize("var n = 1;", Prism.languages[language]);
+(function visit(token: PrismJS.TokenNode): PrismJS.TokenNode {
+    if (typeof token === "string") {
+        return token;
+    } else if (Array.isArray(token)) {
+        return token.map(visit) as PrismJS.TokenNode;
```

Comment:
I need the help of someone with more "type" experience than I to figure out what the correct thing to do here is so that this function doesn't have to cast the return result.